frontend/tests/controllers/UserDeleteTest.php

```diff
@@ -0,0 +1,171 @@
+            new \OmegaUp\Request([
+                'username' => $username,
+                'password' => $username,
+                'email' => $email,
+                'permission_key' => \OmegaUp\Controllers\User::$permissionKey,
+            ])
+        );
+
+        // Check response
+        $this->assertEquals($username, $response['username']);
+
+        // Verify DB
+        $user = \OmegaUp\DAO\Users::FindByUsername($username);
+        $this->assertNotNull($user);
+
+        // Admin will delete $user
+        ['identity' => $admin] = \OmegaUp\Test\Factories\User::createAdminUser();
+
+        // Call api using admin
+        $login = self::login($admin);
+
+        // Call API to request a token
+        $response = \OmegaUp\Controllers\User::apiDeleteRequest(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'username' => $username,
+            ])
+        );
+
+        $this->assertNotNull($response['token']);
+
+        // Call API
+        $response = \OmegaUp\Controllers\User::apiDeleteConfirm(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'username' => $username,
+                'token' => $response['token'],
+            ])
+        );
+
+        // Verify DB user no longer exists
+        $user = \OmegaUp\DAO\Users::FindByUsername($username);
+        $this->assertNull($user);
+    }
+
+    /**
+     * User themselves deletes their account
+     */
+    public function testDeleteUser() {
+        // Creates a user
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+
+        // Verify DB
+        $user = \OmegaUp\DAO\Users::FindByUsername($identity->username);
+        $this->assertNotNull($user);
+
+        // Call api using admin
+        $login = self::login($identity);
+
+        // Call API to request a token
+        $response = \OmegaUp\Controllers\User::apiDeleteRequest(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+            ])
+        );
+
+        $this->assertNotNull($response['token']);
+
+        // Call API
+        $response = \OmegaUp\Controllers\User::apiDeleteConfirm(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'token' => $response['token'],
+            ])
+        );
+
+        // Verify DB user no longer exists
+        $user = \OmegaUp\DAO\Users::FindByUsername($identity->username);
+        $this->assertNull($user);
+    }
+
+    /**
+     * Three User themselves deletes their account
+     */
+    public function testDeletethreeUser() {
+        $identities = [];
+        $numberOfContestants = 3;
+        foreach (range(0, $numberOfContestants - 1) as $id) {
+            [
+            'identity' => $identities[$id],
+            ] = \OmegaUp\Test\Factories\User::createUser();
+
+          // Verify DB
```

Comment:
```suggestion
            // Verify DB
```

can all the comments be aligned with the code they describe?

Reply:
yess! Sure I will aligned them.